hardinfo/usb_util.c

```diff
@@ -279,17 +298,53 @@ static gboolean usb_get_interface_sysfs(int conf, int number,
     if (intf->if_label == NULL)
         intf->if_label = h_sysfs_read_string(ifpath, "interface");
 
+    if (intf->if_class_str == NULL && intf->if_subclass_str == NULL
+                                   && intf->if_protocol_str == NULL) {
+        tmp = g_strdup_printf("C %02x/%02x/%02x", intf->if_class,
+                               intf->if_subclass, intf->if_protocol);
+        scan_ids_file(usb_ids_file, tmp, &result, -1);
+        if (result.results[0])
+            intf->if_class_str = g_strdup(result.results[0]);
+        if (result.results[1])
+            intf->if_subclass_str = g_strdup(result.results[1]);
+        if (result.results[2])
+            intf->if_protocol_str = g_strdup(result.results[2]);
+        g_free(tmp);
+    }
+
     g_free(ifpath);
     return TRUE;
 }
 
+void find_usb_ids_file() {
+    if (usb_ids_file) return;
+    char *file_search_order[] = {
+        g_build_filename(g_get_user_config_dir(), "hardinfo", "usb.ids", NULL),
+        g_build_filename(params.path_data, "usb.ids", NULL),
+        NULL
+    };
+    int n;
+    for(n = 0; file_search_order[n]; n++) {
+        if (!access(file_search_order[n], R_OK))
```

Comment:
There's a bug. Should be `if (!usb_ids_file && !access(file_search_order[n], R_OK))`
The loop needs to complete to free the unused ones, but it should only be set for the first one found.
My fault, I'll have to fix it for arm.ids and sdcard.ids as well.

Reply:
np, I pushed the fix